test-apps/sediment/src/main/java/com/twosigma/waiter/sediment/Servlet.java

```diff
@@ -0,0 +1,222 @@
+package com.twosigma.waiter.sediment;
+
+import com.google.gson.Gson;
+import org.eclipse.jetty.http.HttpFields;
+import org.eclipse.jetty.server.HttpConfiguration;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.Response;
+
+import javax.servlet.ServletException;
+import javax.servlet.ServletOutputStream;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class Servlet extends HttpServlet {
+
+    private final static Logger LOGGER = Logger.getLogger(Servlet.class.getName());
+
+    static String retrieveCorrelationId(HttpServletRequest request) {
+        final Enumeration<String> headerNames = request.getHeaderNames();
```

Comment:
Can we use `request.getHeader("x-cid")`?

Reply:
Done.